### PR TITLE
[Serve] Fix serve metrics test

### PR DESCRIPTION
**Pull request by Hankpipi**
## Why are these changes needed?

1. `do_assert ` is always `False`, so this test case is meaningless.
2. Some metrics have renamed.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_metrics.py [hunk 1]
```diff
@@ -31,21 +33,19 @@ def verify_metrics(do_assert=False):
...
+            "serve_num_http_requests",
+            "serve_deployment_queued_queries",
+            "serve_deployment_request_counter",
+            "serve_deployment_replica_starts",
             # histogram
             "deployment_processing_latency_ms_bucket",
             "deployment_processing_latency_ms_count",
             "deployment_processing_latency_ms_sum",
             # gauge
             "replica_processing_queries",
```
Comment (shrekris-anyscale) [inline python/ray/serve/tests/test_metrics.py:46]: This metric is now called `"serve_replica_processing_queries"`. Can we change this string to that?

File: python/ray/serve/tests/test_metrics.py [hunk 2]
```diff
@@ -31,21 +33,19 @@ def verify_metrics(do_assert=False):
...
-            "num_router_requests_total",
-            "num_http_requests_total",
-            "deployment_queued_queries_total",
-            "deployment_request_counter_requests_total",
-            "deployment_worker_starts_restarts_total",
+            "serve_num_router_requests",
+            "serve_num_http_requests",
+            "serve_deployment_queued_queries",
+            "serve_deployment_request_counter",
+            "serve_deployment_replica_starts",
```
Comment (shrekris-anyscale) [inline python/ray/serve/tests/test_metrics.py:40]: Can we also check for `"serve_deployment_processing_latency_ms"` here?

**Comment by shrekris-anyscale** [comment]
Nice work! Can we rename the unit test to `test_serve_metrics_for_successful_connection`? None of the error metrics are present in the response because we're only testing a successful connection.

**Comment by Hankpipi** [comment]
Hi @shrekris-anyscale, I have committed the changes according to your suggestions.

**Review by shrekris-anyscale — APPROVED** [review]
Great, thanks for making the changes! Everything looks good to me.

**Pull request merged**